apiserver/common/imagecommon/imagemetadata.go

```diff
@@ -13,29 +15,30 @@ import (
 	"github.com/juju/juju/state/cloudimagemetadata"
 )
 
-// ImageMetadataInterface is an interface for manipulating images metadata.
-type ImageMetadataInterface interface {
-
+// ImageMetadataSaver is an interface for manipulating images metadata.
+type ImageMetadataSaver interface {
 	// SaveMetadata persists collection of given images metadata.
 	SaveMetadata([]cloudimagemetadata.Metadata) error
-
-	// ModelConfig retrieves configuration for a current model.
-	ModelConfig() (*config.Config, error)
 }
 
 // Save stores given cloud image metadata using given persistence interface.
-func Save(st ImageMetadataInterface, metadata params.MetadataSaveParams) ([]params.ErrorResult, error) {
+func Save(
```

Comment:
You might be right. It looks like this was pulled out in #7631 in prep for some future work, I'm not sure if that future work ever happened though.

Reply:
Right, so my reading of the history is:
- #7631 pulled this code into a common package, in preparation for #7636 which created a new controller-side image metadata facade
- #8400 essentially removed the controller-side facade, so since then, this code has only been used by the client facade.

In that case, I guess we're good to move this back into the `client/imagemetadatamanager` package, right @SimonRichardson @wallyworld ?

Reply:
Agree with @nvinuesa - even if it's not exactly the right place, common packages that aren't common are anti-pattern. We should move them, we can always move them in the future.